docs/source/developers/cpp/building.rst

```diff
@@ -145,45 +238,34 @@ Minimal release build (1GB of RAM for building or more recommended):
 
 .. code-block:: shell
 
-   git clone https://github.com/apache/arrow.git
-   cd arrow/cpp
-   mkdir release
-   cd release
-   cmake ..
-   make
+   $ mkdir build-release
```

Comment:
I know you added on top that all instructions assume you navigated to `arrow/cpp`, but I think it might be worth repeating this a few times at least when creating a new directory (eg could add a comment on the same line like `# inside arrow/cpp directory`)

Reply:
+1